hazelcast/src/main/java/com/hazelcast/dataconnection/impl/JdbcDataConnection.java

```diff
@@ -156,13 +162,14 @@ public Connection getConnection() {
 
     @Override
     public void destroy() {
-        if (pooledDataSource != null) {
+        if (pooledDataSourceSup != null) {
+            HikariDataSource dataSource = pooledDataSourceSup.get();
```

Comment:
if destroy is called 2 times from different thread, pooledDataSourceSup != null may be non-null and then null in other line. You should introduce a local variable for pooledDataSourceSup.

Reply:
can that happen? destroy should never be invoked more than once according to logic in `ReferenceCounter`

Reply:
ok, it can happen possibly happen when user drops a data link when member is being shut down